hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -98,17 +106,20 @@ public MemStoreCompactor(CompactingMemStore compactingMemStore) {
    * is already an ongoing compaction or no segments to compact.
    */
   public boolean start() throws IOException {
-    if (!compactingMemStore.hasImmutableSegments()) return false;  // no compaction on empty
+    if (!compactingMemStore.hasImmutableSegments()) { // no compaction on empty pipeline
+      return false;
+    }
 
-    int t = compactingMemStore.getConfiguration().getInt(COMPACTING_MEMSTORE_TYPE_KEY,
+    String memStoreType = compactingMemStore.getConfiguration().get(COMPACTING_MEMSTORE_TYPE_KEY,
         COMPACTING_MEMSTORE_TYPE_DEFAULT);
 
-    switch (t) {
-      case 1: type = Type.COMPACT_TO_SKIPLIST_MAP;
+    switch (memStoreType) {
+      case "index-compaction": action = Action.MERGE;
         break;
-      case 2: type = Type.COMPACT_TO_ARRAY_MAP;
+      case "data-compaction": action = Action.COMPACT;
```

Comment:
even if the user defines "data-compaction" we might still take another action.
it's better to save the user request as the starting point but set the action before we actually do it.
Again this is something I would defer to the next Jira, but should do here if there's no Jira